ts41/join.d.ts

```diff
@@ -0,0 +1,22 @@
+/**
+Joins an array of strings using a given string as delimiter.
+
+Use-case: Defining field paths in a nested object (e.g. for dot-notation fields in MongoDB queries).
+
+@example
+```
+import {Join} from 'type-fest';
+
+const path: Join<['foo', 'bar', 'baz'], '.'> = ['foo', 'bar', 'baz'].join('.');
+```
+
+@category Template Literals
+*/
+export type Join<
+	Strings extends string[],
+	Delimiter extends string
+> = Strings extends [] ? '' :
+	Strings extends [string] ? `${Strings[0]}` :
+	// @ts-expect-error `Rest` is inferred as `unknown` here
```

Comment:
Is this a TS limitation or bug?

Reply:
I opened https://github.com/microsoft/TypeScript/issues/45281 for this.

If this is confirmed as intentional and not a TS bug, I'll update the generic type in this type to `unknown[]` instead of `string[]`.

Reply:
Can you add a link to that issue as a code comment?